extensions/ql-vscode/src/remote-queries/sarif-processing.ts

```diff
@@ -163,17 +169,45 @@ export function tryGetRule(
 }
 
 function getCodeSnippet(
+  contextRegion?: sarif.Region,
   region?: sarif.Region,
-  alternateRegion?: sarif.Region,
 ): CodeSnippet | undefined {
-  region = region ?? alternateRegion;
+  const actualRegion = contextRegion ?? region;
 
-  if (!region) {
+  if (!actualRegion) {
     return undefined;
   }
 
-  const text = region.snippet?.text || "";
-  const { startLine, endLine } = parseSarifRegion(region);
+  const text = actualRegion.snippet?.text || "";
+  const { startLine, endLine } = parseSarifRegion(actualRegion);
+
+  if (
+    contextRegion &&
+    region &&
+    text.length > CODE_SNIPPET_LARGE_LINE_SIZE_LIMIT
+  ) {
+    const code = text.split("\n");
+
+    const highlightedRegion = parseSarifRegion(region);
+
+    const highlightedLines = code.map((line, index) => {
+      return parseHighlightedLine(line, startLine + index, highlightedRegion);
+    });
+
+    const highlightedCharactersCount = highlightedLines
+      .map((line) => line.highlightedSection.length)
+      .reduce((a, b) => a + b, 0);
```

Comment:
```suggestion
    const highlightedCharactersCount = highlightedLines.reduce(
      (total, line) => total + line.highlightedSection.length,
      0,
    );
```
Minor point but we could avoid the `map` step to avoid iterating the list twice. I don't think it makes the code less clear in this case.